blockchain/chain/src/store/chain_store.rs

```diff
@@ -210,6 +215,18 @@ where
         }
         Ok(())
     }
+
+    /// Gets 32 bytes of randomness for ChainRand paramaterized by the DomainSeparationTag, ChainEpoch, Entropy
+    pub fn get_randomness(
+        &self,
+        blocks: &TipsetKeys,
+        pers: DomainSeparationTag,
+        round: ChainEpoch,
+        entropy: &[u8],
+    ) -> Result<[u8; 32], Error> {
+        get_randomness(self.db.as_ref(), blocks, pers, round, entropy)
+            .map_err(|e| Error::Other(e.to_string()))
+    }
```

Comment:
Things are going to get pretty messy and confusing if we duplicate all of these, any specific reason to have it like this? Another option is to make the get_randomness function associated but use a ref to blockstore instead of `&self`

```suggestion
```

Reply:
Ill remove, I agree